### PR TITLE
Do not break DROP DICTIONARY with DROP TABLE

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Do not break DROP DICTIONARY with DROP TABLE (otherwise after this error you will not be able to remove database)

### CODE REVIEW

File: src/Interpreters/InterpreterDropQuery.cpp [hunk 1]
```diff
@@ -112,6 +112,9 @@ BlockIO InterpreterDropQuery::executeToTable(
             context.checkAccess(table->isView() ? AccessType::DROP_VIEW : AccessType::DROP_TABLE, table_id);
             table->checkTableCanBeDropped();
 
+            if (database->isDictionaryExist(context, table_name))
```
Comment (vitlibar) [inline src/Interpreters/InterpreterDropQuery.cpp:115]: It seems this check might be moved into `table->checkTableCanBeDropped()`, i.e. into the function `StorageDictionary::checkTableCanBeDropped()`.

File: src/Databases/DatabasesCommon.cpp [hunk 2]
```diff
@@ -109,6 +107,11 @@ void DatabaseWithOwnTablesBase::attachTableUnlocked(const String & table_name, c
     }
 }
 
+bool DatabaseWithOwnTablesBase::isDictionaryExist(const Context & /*context*/, const String & name) const
```
[the remark this answers is not in the document]
Comment (azat) [reply src/Databases/DatabasesCommon.cpp]: Indeed, thanks, fixed!

File: src/Interpreters/InterpreterDropQuery.cpp (hunk 1 above)
Comment (azat) [reply src/Interpreters/InterpreterDropQuery.cpp]: Looks better, thanks! Fixed (and added `backQuoteIfNeed`)

File: src/Storages/StorageDictionary.cpp [hunk 3]
```diff
@@ -45,6 +47,11 @@ StorageDictionary::StorageDictionary(
     }
 }
 
+void StorageDictionary::checkTableCanBeDropped() const
+{
+    throw Exception("Cannot detach dictionary " + backQuoteIfNeed(dictionary_name) + " as table, use DETACH DICTIONARY query.", ErrorCodes::UNKNOWN_TABLE);
+}
```
Comment (HesyRa) [inline src/Storages/StorageDictionary.cpp:53]: @azat Hello. I've just tried new build `ClickHouse server version 20.4.5.36 (official build).` and I can't delete Table with Engine=Dictionary. Always have a exception to use DETACH DICTIONARY when I try to remove whole database...

I think here can be an error. Can you help me?

**Pull request merged**